service_capacity_modeling/interface.py

```diff
@@ -895,39 +895,52 @@ def reference_shape(self) -> Instance:
         return default_reference_shape
 
     def merge_with(self, defaults: "CapacityDesires") -> "CapacityDesires":
-        # Now merge with the models default
-        desires_dict = self.model_dump()
         default_dict = defaults.model_dump()
+        desires_dict = self.model_dump()
 
-        default_dict.get("query_pattern", {}).update(
-            desires_dict.pop("query_pattern", {})
-        )
-        default_dict.get("data_shape", {}).update(desires_dict.pop("data_shape", {}))
```

Comment:
These lines are just throwing away the existing default response in the first get if it does not exist.